### DISCUSSION TITLE
Apple TV Beta Testing Group

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#200] [Announcements]
# Upcoming Apple TV Compatibility
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/34d4f666-f44f-4224-908c-9464f9c5faf6)

I'd love to have **YOU** as a beta tester!  Follow this thread and you'll be notified right away when new versions are ready, and you'll be first in line to share your input before launch!

## To do list
  - [x] Publish pre-release version v2.4.0-b1
  - [x] Publish pre-release version v2.4.0-b2
  - [x] Publish pre-release version v2.4.0-b3
  - [x] Publish pre-release version v2.4.0-b4
  - [x] Publish official release version 3.0.0
  - [x] Create a translation mechanism for Apple TV App names that do not remain in English when the Apple TV's default language is something different
  - [x] Create and verify translations to 'nl' (Dutch)
  - [x] Create Apple Remote style 1
  - [x] Create Apple Remote style 2
  - [x] Fix the remote style 2 app layout bug with "default" remote style selected using "appletv-gen4" device type (will be fixed in beta 4)
  - [x] Change the default number of app launch buttons on style 2 from 6 to 8
  - [x] Test & Fix functionality when using an Apple TV device with the AL2 remote style
  - [x] Work out some misc Card Configuration UI/UX
  - [x] Add AppleTV devices to the stub config for better on-boarding of users who only have Apple TV entities
  - [ ] Consider a toggle option to replace the mute button with a control center launch button
  - [x] Partner with all other beta testers that speak a language different from English, and work out proper translations for apple apps and setup menu headings
  - [x] Update documentation

## Known Limitations / Issues
 - The mute button DOES NOT WORK without using a [button override](https://github.com/PRProd/HA-Firemote#button-overrides)
 - There does not seem to be a way to command a restart
 - The Home Assistant Apple TV Integration does not provide a way to know which app is currently active, so I an unable to make the app shortcut buttons take on a 'lit' or 'dim' state like the android integration remotes can
 - If your Apple TV language is not set to English, it's possible that not all of the native Apple apps will launch with the shortcuts.  With your help, this CAN be fixed.  Please post below!

**Comment by jrspowers** [comment]
Hey, I'd also like to participate in testing; I have an Apple TV at home too. Looking forward to it! 😊

**Reply by PRProd** [reply]
@jrspowers - Which model do you own?

**Reply by jrspowers** [reply]
> @jrspowers - Which model do you own?

Hey, I have the Apple TV HD here, and I also have the Apple TV 4K, but I can't say the exact model year as it's not with me right now. However, I can always get it to test for you. 😊![image](https://github.com/PRProd/HA-Firemote/assets/60328474/0ba3a505-40ec-4b44-ae4f-3b87e82d10e9)

**Comment by Paradox8** [comment]
Very excited! Lock me in for Beta testing. I have Apple TV 4K 2022

**Comment by PRProd** [comment]
# Beta 1 has been released
attn: @jrspowers @Paradox8 and any others
Good news everyone!  The Apple TV Beta Release version [v2.4.0-b1 ](https://github.com/PRProd/HA-Firemote/releases/tag/v2.4.0-b1) has just been published for your review.
<br>

## Instructions for beta testers
I have updated the original post at the top of this discussion, and I will continue to do so as we move forward.  There, you will find the current status of this new device functionality.
 - Please carefully review the **Known Limitations / Issues** section before reporting issues or making suggestions.
 - Please reply to THIS post, and not the discussion itself for specific comments about this specific beta release
 - Please provide any/all feedback, both positive and negative.  I need to know what works and what doesn't work.
<br>

I hope you enjoy it!  I look forward to hearing from you all!

**Reply by jrspowers** [reply]
Hey, I'm going to test this for you in HD and 4K versions shortly. Very nice to see this 😊

**Reply by jrspowers** [reply]
Hey, I just tested the Apple TV 4 HD, and it works very nicely, but there's one function missing for changing apps, right? Also, I can say that to fully close apps, you can click the AirPlay logo twice, then navigate to the app you want to close using the arrow keys, and finally press the upper arrow key twice to close it. So, an extra button for this isn't necessary. 

I'm missing a few apps that I thought would be there, but I don't see them for the Apple TV. These apps are vtmgo Belgium app, streamz Belgium app, and Proximus pickx app. Could you add these? 

I've tested the apps that are available, After restarting the Apple TV, Videoland works, but now the settings button & photo doesn't work anymore?. Still, great job overall 😊![image](https://github.com/PRProd/HA-Firemote/assets/60328474/c52c7b69-4383-47c4-9b10-d6e4c9c405fe)

**Reply by PRProd** [reply]
Moving @jrspowers comments to the proper thread:

> Hey,
> I saw that you had a problem with not being able to launch the App Store. It's strange that it's in the list of sources, and it opens nicely, so why can't you use it? Or am I mistaken?

> Hey,
>
>Would it also be possible to remove the mute button and replace it with another button, like the Control Center? That would be useful.

>Hey, I just tested the Apple TV 4K version, and it works well. However, I'm facing the same issue here where the settings launch doesn't work, and the buttons for music, photos, FaceTime, movies, search, and TV shows don't work either. The rest of the buttons work fine, so hopefully, you can fix this and address the previously asked questions too? 😊

**Reply by jrspowers** [reply]
Hey, did I put it under the right category, or what do you mean?

**Reply by PRProd** [reply]
@jrspowers - I'm just trying to keep the conversation clean.  Your replies all quoted the original post, so I've been removing the quoted sections to keep it clean.  Also, 3 of your replies were posted into the root of the discussion itself, and not this thread about the beta release 1, so I moved those.

**Reply by jrspowers** [reply]
> @jrspowers - I'm just trying to keep the conversation clean.  Your replies all contain quotes from the previous post, so I've been removing them to keep it clean.  3 of your replies went to the root of the discussion itself, and not this thread about the beta release 1, so I moved those.

Yes, I understand, but what did you move, and what, in your opinion, has nothing to do with this? What I mentioned was the request for apps that were not there, but otherwise, it was all about this Apple TV beta, right? But where can I find my other responses, or did you delete them completely?

**Reply by PRProd** [reply]
@jrspowers - THANKS FOR THE FEEDBACK!  I sincerely appreciate it! :smile: 

Here are my answers to the questions you've asked.  Hopefully this can keep us moving forward.
<br>

> Would it also be possible to remove the mute button and replace it with another button, like the Control Center? That would be useful.

I thought about doing that, but ultimately chose to keep the mute button.  I use that button often, and I'd really like to keep this as close to the original look of the physical remote as possible.  If others agree with you on switching it with something else, you and the community might be able to change my mind

<br>

> I saw that you had a problem with not being able to launch the App Store. It's strange that it's in the list of sources, and it opens nicely, so why can't you use it? Or am I mistaken?

I agree.  It is strange.  All of the other Apple Apps (photos, podcasts, etc.) worked by just calling them using their names found in the list of sources, but for whatever reason the App Store did not.  I'm not done trying, I'm just temporarily stuck on that one and needed to point it out as a known issue before releasing the beta.
<br>

> I'm missing a few apps that I thought would be there, but I don't see them for the Apple TV. These apps are vtmgo Belgium app, streamz Belgium app, and Proximus pickx app. Could you add these?

Yes.  Not everything was available in the US app store, so I only brought over what I could test.  I'll need your help with this, and those instructions are below.
<br>
<br>

## Apple TV - Gen 4 2014
I don't own this device, so your feedback is VERY important to me here!
<br>
  > there's one function missing for changing apps, right?
  
That function does exist!.
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/968c9df9-9714-4f69-ba50-e7dd88fc91d7)
 >  you can click the AirPlay logo twice, then navigate to the app you want to close using the arrow keys, and finally press the upper arrow key twice to close it. So, an extra button for this isn't necessary.
 
 Actually, it is necessary.  Unfortunately, I am unable to / don't know how to properly capture a double click or a push and hold on the Firemote card, so anything you normally do by double clicking or holding on the physical remote has to be built as a separate button inside of Firemote.  That's why I made that option into a "function" button.
<br>

> I've tested the apps that are available, After restarting the Apple TV, Videoland works, but now the settings button & photo doesn't work anymore?

I noticed something strange when downloading new apps and trying to launch them through Home Assistant.  Even if I knew the App's proper name, it wouldn't launch because the new app hadn't yet been added to the source list.  I had to reload the Apple TV Integration first to get the new app added to the list, and then it would launch.  I'm wondering if that's also true in your situation as well.  Could you do the following steps for me?
 - Open the Devices and Services Menu: 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b3e27a3a-011e-443d-92a5-3b031144b69a)
 - Click on the > next to Apple TV 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/aadba8bd-bb8a-4e38-9047-3b900c792266)
 - Next, click the three dots next to CONFIGURE, and select Reload from the menu 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/a9b8cab3-eb05-4dda-92ff-99273227952b)
 - After the reload process has completed, open the developer tools menu, and select the STATES tab.  Here, we will need to look at your Apple TV - Gen 4 2014 device.  You can scroll around until you find it, or you could start typing the name of it beginning with "media_player." to help you filter through the entity list.
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/ff661446-2639-4854-9c94-d966452fe060)
 - Look through the "source_list" in the Attributes column to see what is available.  If you see entries for "Settings" and "Photos", then it's possible that the Firemote buttons for those apps will work again.  Either way, in order to best help you with your new & missing app requests, please copy/paste the text from that field here in a reply.  For instance, mine looks like this:
 ```
source_list: AMC+, App Store, Arcade, ARD Mediathek, ARTE, bgtime.tv, Channels, CNN, Computers, Crunchyroll, CuriosityStream, Daily Wire, DAZN, DIRECTV, discovery+, Disney+, DS video, Emby, ESPN, F1 TV, FaceTime, Fitness, FOX Sports, Freevee, gcn, HDHomeRun, Hulu, IPVanish, iQIYI, JustWatchTV, Magenta TV, Max, Mediaset Infinity, MLB, Movies, Music, NBC Sports, Nebula, Netflix, NFL, NLZIET, NPO, OQEE by Free, Pandora, Paramount+, PBS, Peacock, Photos, Plex, Pluto TV, Podcasts, Prime Video, RaiPlay, Search, Settings, ShopHQ, Showtime, SiriusXM Radio for TV, Sky News, Speedtest, Spotify, STARZ, Steam Link, Stream, SVT Play, TBS, Tennis Channel, TIDAL, TNT, Tubi, TV, TV Shows, TVNZ+, Twitch, VBTV, Viaplay, Videoland, VLC, VRT MAX, XCIPTV, YouTube, YouTube Kids, YouTube TV, ZDFmediathek
```

<br>

## Apple TV 4K - Gen 2 2021
I do own this device, so it should be easier to discuss and resolve issues you see here.
<br>

> I just tested the Apple TV 4K version, and it works well. However, I'm facing the same issue here where the settings launch doesn't work, and the buttons for music, photos, FaceTime, movies, search, and TV shows don't work either. The rest of the buttons work fine

I have two thoughts about this.  First, if you reload the integration as described above, does that make a difference?  Second, I wonder if it could be a language issue.  Are the settings, music, photos, FaceTime, etc. named exactly that way in your source_list, or are they different in your language?  If they are different, I'd love to get a copy/paste of your source list on this 4K Gen 2 2021 device as well.
<br>
<br>
<br>
THANK YOU AGAIN for your testing!  I really do appreciate your time!

**Reply by PRProd** [reply]
> Yes, I understand, but what did you move, and what, in your opinion, has nothing to do with this? What I mentioned was the request for apps that were not there, but otherwise, it was all about this Apple TV beta, right? But where can I find my other responses, or did you delete them completely?

There really isn't a good way to move or re-organize a discussion thread here.  What I did was copy your posts that ended up in the root of the discussion, and pasted them into a new reply inside this **Beta 1 has been released** thread.  Once they had been copied and pasted, I did delete the discussion root posts, but NOTHING  was lost, just copied and pasted into a different place.  I appreciate 100% of your feedback, and it is ALL important and relevant!  I'm sorry if my re-arranging accidentally upset you in any way, that was not my intent.

**Reply by jrspowers** [reply]
> > Yes, I understand, but what did you move, and what, in your opinion, has nothing to do with this? What I mentioned was the request for apps that were not there, but otherwise, it was all about this Apple TV beta, right? But where can I find my other responses, or did you delete them completely?
> 
> 
> 
> There really isn't a good way to move or re-organize a discussion thread here.  What I did was copy your posts that ended up in the root of the discussion, and pasted them into a new reply inside this **Beta 1 has been released** thread.  Once they had been copied and pasted, I did delete the discussion root posts, but NOTHING  was lost, just copied and pasted into a different place.  I appreciate 100% of your feedback, and it is ALL important and relevant!  I'm sorry if my re-arranging accidentally upset you in any way, that was not my intent.

I definitely understand what you're saying, no problem at all that you did this. And I can give you a tip: consider creating a Discord server for your testing. It's much easier for communication, and you can use it for various purposes and keep track of things more efficiently.

**Reply by PRProd** [reply]
@jrspowers - Thanks for the tip!  I'll look into it!

**Reply by PRProd** [reply]
@jrspowers - I've never used discord, other than looking at it for about 5 minutes a few years ago.  Can you help me get organized over there?  I think I need to share this link with you?  https://discord.gg/Ycc8wcBy

**Comment by jrspowers** [comment]
> - BEDANKT VOOR DE FEEDBACK! Ik waardeer het oprecht! 😄
> 
  You're welcome!
> Hier zijn mijn antwoorden op de vragen die je hebt gesteld. Hopelijk kan dit ons vooruit helpen.
> 
We're going to try this, right?

> Ik dacht erover om dat te doen, maar koos er uiteindelijk voor om de mute-knop te behouden. Ik gebruik die knop vaak en ik wil deze graag zo dicht mogelijk bij het oorspronkelijke uiterlijk van de fysieke afstandsbediening houden. Als anderen het met je eens zijn om het met iets anders te veranderen, kunnen jij en de gemeenschap misschien van gedachten veranderen

I understand that you don't want to remove the mute button, but is there a possibility to change it so that users can choose between the mute or control center button? This would make it more interesting for me.
> Ik ben het ermee eens. Het is vreemd. Alle andere Apple-apps (foto's, podcasts, enz.) werkten door ze gewoon te bellen met hun namen in de lijst met bronnen, maar om welke reden dan ook deed de App Store dat niet. Ik ben nog niet klaar met proberen, ik zit gewoon tijdelijk vast aan die ene en moest het aanwijzen als een bekend probleem voordat ik de bèta vrijgaf.

Yes, it's definitely a very strange issue with that app store. Hopefully, you can still manage to fix it.
 > Ja. Niet alles was beschikbaar in de Amerikaanse app store, dus ik nam alleen mee wat ik kon testen. Ik heb je hulp hierbij nodig, en die instructies staan hieronder.

I'm happy to help. Here are the source names of the apps I would like to see added that you cannot install in your app store. So, if you need more information about this, I'd be happy to share it.

 ```
source_list: IPTVX, Proximus Pickx, Qmusic-BE, Streamz, TestFlight, VTM GO, 
 ```


> ## Apple TV - Gen 4 2014
> Ik heb dit apparaat niet, dus uw feedback is erg belangrijk voor mij hier!
> 
> > Er ontbreekt één functie voor het wijzigen van apps, toch?

What I actually mean is that if you hold down on an app using the physical remote, you get the option to change its position, move it to the home screen, or delete the app. Do you think it's possible to add this functionality or do you think it's not feasible?

> Eigenlijk is het nodig. Helaas kan / weet ik niet hoe ik een dubbelklik of een druk en houd op de Firemote-kaart goed vast, dus alles wat je normaal doet door te dubbelklikken of de fysieke afstandsbediening vast te houden, moet als een aparte knop in Firemote worden gebouwd. Daarom heb ik van die optie een "functie" knop gemaakt.

Look, if you double-click the home button, you'll access the same menu that you do with the "function: switch apps." You can navigate to the left using your arrow keys, and when you're on an app, you can delete it by double-clicking the up arrow key twice in a row. This completely closes the app, and it works perfectly. Hopefully, you understand it better this way because sending a video is difficult.

> Ik merkte iets vreemds op bij het downloaden van nieuwe apps en probeerde ze te starten via Home Assistant. Zelfs als ik de juiste naam van de app zou weten, zou deze niet worden gestart omdat de nieuwe app nog niet aan de bronlijst was toegevoegd. Ik moest eerst de Apple TV-integratie opnieuw laden om de nieuwe app aan de lijst toe te voegen en vervolgens zou deze worden gestart. Ik ben benieuwd of dat in jouw situatie ook zo is. Kunt u de volgende stappen voor mij uitvoeren?

I did this, but it doesn't solve the problem. But as you mentioned below, it does indeed have to do with the language. My language is Dutch, and then it has a different name. Here is the source list in my language; hopefully, you can add this and it will fix the problem.

```
source_list: Amazon Music, Apple Events, App Store, Arcade, Cloud Stream TV, Computers, dailymotionTV, DayView, Disney+, Doodle Jump Race, Films, Flow, Foto's, GSE IPTV, iHomeTouch, Instellingen, IPTVX, KodiTVBox, Looming, Muziek, myTuner Radio Free, Netflix, NPO, PAC-MAN PARTY, Plex, Podcasts, Popcorn Time, Prime Video, Proximus Pickx, Qmusic-BE, Speedtest, Spotify, Steam Link, Streamz, TestFlight, TV, tvOS Browser, Videoland, VLC, VRT MAX, VTM GO, YouTube, Zoek
```
I hope this is enough information. If you need anything else, feel free to ask; I'd be happy to assist you thoroughly!

**Comment by PRProd** [comment]
# Beta 2 Has Been Released
attn: @jrspowers @Paradox8 and any others

Good news everyone! The Apple TV Beta Release version [v2.4.0-b2 ](https://github.com/PRProd/HA-Firemote/releases/tag/v2.4.0-b2)has just been published for your review.
<br>

## Instructions for beta testers
I have updated the original post at the top of this discussion, and I will continue to do so as we move forward. There, you will find the current status of this new device functionality.
 - Please carefully review the Known Limitations / Issues section before reporting issues or making suggestions.
 - Please reply to THIS post, and not the discussion itself for specific comments about this specific beta release
 - Please provide any/all feedback, both positive and negative. I need to know what works and what doesn't work.
 
 I look forward to hearing from you all!  The more information you provide, the better this release will be!  You can comment here on GitHub, or head over to my discord server and [join the conversation there](https://discord.gg/Ycc8wcBy).

**Comment by PRProd** [comment]
# Beta 3 Has Been Released
attn: @jrspowers @Paradox8 and any others

Good news everyone! The Apple TV Beta Release version [v2.4.0-b3 ](https://github.com/PRProd/HA-Firemote/releases/tag/v2.4.0-b3)has just been published for your review.
<br>
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/295caefd-c54b-42b4-bfc2-97dea1eef272)

## What's new
 - EN to NL (English to Dutch) translations are now also on the card config page
 - Apple TV Remote Style 1 is available
 - Apple TV Remote Style 2 is available
 - Apple TV (HD) Gen4 2015 device is now also enabled from the menu

## Instructions for beta testers
I have updated the original post at the top of this discussion, and I will continue to do so as we move forward. There, you will find the current status of the Firemote beta and Apple TV functionality.
 - Please carefully review the Known Limitations / Issues section before reporting issues or making suggestions.
 - Please reply to THIS post, and not the discussion itself for specific comments about this specific beta release
 - Please provide any/all feedback, both positive and negative. I need to know what works and what doesn't work.
 
 I look forward to hearing from you all!  The more information you provide, the better this release will be!  You can comment here on GitHub, or head over to my discord server and [join the conversation there](https://discord.gg/Ycc8wcBy).

**Comment by PRProd** [comment]
# Beta 4 Has Been Released
attn: @jrspowers @Paradox8 and any others

Good news everyone! The Apple TV Beta Release version [v2.4.0-b4 ](https://github.com/PRProd/HA-Firemote/releases/tag/v2.4.0-b4)has just been published for your review.
<br>

## What's new
 - App Store launch button is now available
 - Fixed app layout issue with appletv-gen4 and default remote style selected
 - Various on-boarding and setup card changes
 - Increased max app launch buttons on AR2 from 6 to 8
 - Changed app button lit states in AL1 and AL2 to always be lit when the device being controlled is an Apple TV
 - Changed visible control buttons in AL2 to make more sense when the device being controlled is an Apple TV
 - Added app NOSTV ( https://github.com/PRProd/HA-Firemote/issues/204 @horamarques )
 - Added app TV4 Play ( https://github.com/PRProd/HA-Firemote/issues/202 @Skranta2 )

## Instructions for beta testers
I have updated the original post at the top of this discussion, and I will continue to do so as we move forward. There, you will find the current status of the Firemote beta and Apple TV functionality.
 - Please carefully review the Known Limitations / Issues section before reporting issues or making suggestions.
 - Please reply to THIS post, and not the discussion itself for specific comments about this specific beta release
 - Please provide any/all feedback, both positive and negative. I need to know what works and what doesn't work.
 
 I'm hoping that this is the last beta version before the official release, so speak up now if you have found an issue that hasn't been corrected yet, or isn't on the checklist yet.  Please comment here on GitHub, or head over to my discord server and [join the conversation there](https://discord.gg/Ycc8wcBy).

**Comment by PRProd** [comment]
# Firemote Version 3 has just been released!
The Apple TV version of Firemote has just been officially released as [v3.0.0](https://github.com/PRProd/HA-Firemote/releases/tag/v3.0.0)! :tada: 

THANK YOU so much to @jrspowers for all of your help with testing and discussions!  I couldn't have done it without you!

**Reply by PRProd** [reply]
I posted an announcement here as well: https://community.home-assistant.io/t/firemote-card-does-more-it-now-controls-apple-tv-devices-too/627918

**Reply by PRProd** [reply]
Here too: https://www.reddit.com/r/homeassistant/comments/17a9ntc/firemote_card_does_more_it_now_controls_apple_tv/

**Reply by Paradox8** [reply]
I've just had a quick dive through the remote - my Apple TV obviously isn't an officially supported version - Apple TV 4K Gen3 2022 (Model - A2843). I'm running the very latest tvOS v17 (released September 18th)

First thing I want to say is wow. Super responsive, cannot notice any difference between using Firemote and the actual remote! You've done an amazing job PRProd.

Navigation - up/down/left/right/enter - works perfectly.
Back - works.
Home - works.
Play/Pause - works.
Volume - doesn't appear to work.
Mute - doesn't appear to work.
Switch Apps - works.
Control Panel - works.
Skip Forward/Back - works.
Next/Previous - only tested with Plex so not sure if these buttons are compatible, but they didn't appear to do anything.
App Shortcuts - tested the following App Store / Search / Settings / Plex / Youtube - none of these work. I get the following error:

_Failed to call service media_player/select_source. Entity media_player.apple_tv does not support this service_


I was reading the conversation history here and had a look at the states section for the apple tv entity in the developer tools, my entity doesn't show any of the sources at all in the attributes column. It simply has the following:

friendly_name: Master Bedroom Apple TV
supported_features: 448439

**Reply by PRProd** [reply]
@Paradox8 - Excellent troubleshooting!  Yes, the missing source list is exactly why those things are not working for you.  Did you try the instructions I left in [that thread](https://github.com/PRProd/HA-Firemote/discussions/200#discussioncomment-7203747) for reloading the Apple TV integration?

**Reply by Paradox8** [reply]
Yes, I did that and still they did not show up. I even tried completely rebooting HA and then trying, as well as rebooting Apple TV and trying again.

I even just tried again then for good measure but still no success!

What version of tvOS is your Apple TV running? Maybe its version related? I've just had a look through the known issues for the Apple TV integration and nothing mentioned - but I suppose tvOS 17 is barely a month old so there is a possibility its a bug/limitation of the new OS version and no one has noticed yet.

**Reply by PRProd** [reply]
@Paradox8 ~
I'm running tvOS 17 also.
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/35c554c7-81f2-4e4e-bf8d-f1bf41657996)
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/4823673f-d78f-44b0-90a5-85204d266340)

I have another thought.  Is it possible that you have an early, test, or HACS version of the Apple TV integration?  If you do, or even if you don't, I wonder what would happen if you completely remove your Apple TV from HA, and then re-add it.

**Reply by Paradox8** [reply]
Sorry for taking a few days to get back to you. Been working very long days.
Removed integration and set back up and voila, I'm seeing the source_list now.

Only thing left that isn't working is the volume controls - but I'm assuming these won't work for anyone as on the normal remote it would actually be doing this via IR direct to TV rather than going via the Apple TV itself.....

Which brings me to a feature suggestion - ability to customise the volume buttons on the remote so it controls the volume for a different entity - i.e. media_player.samsung_tv - this way, that apple tv firemote could do all functions.

**Reply by PRProd** [reply]
@Paradox8 - YAY!  I'm glad it's working now!  

The volume buttons will work through Firemote if you are using CEC.  The mute button will not work at all without using a button override... which happens to be the feature you suggested, which already exists!  The volume buttons can also use button overrides if you don't want to go the CEC route.

Take a look at the [Why won't the volume, mute, and/or power buttons work from my Firemote card?](https://github.com/PRProd/HA-Firemote#why-wont-the-volume-mute-andor-power-buttons-work-from-my-firemote-card) for a quick overview.  Of course, I'm more than happy to help you out if you get stuck.  If the documentation is unclear, please let me know.  I'm always looking to improve.

**Comment by Paradox8** [comment]
OMG! I'm so sorry I didn't help with testing! I don't know why but I wasn't getting alerts even though I have them enabled?!? I just saw the release in HACS! You're amazing!

EDIT: Turns out I was getting the notifications. They were all going to my spam folder.

Thanks for all your hard work getting this up and running guys!

**Reply by PRProd** [reply]
Silly spam filters  :upside_down_face:
